spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -648,6 +650,21 @@ public static org.apache.iceberg.Table loadIcebergTable(SparkSession spark, Stri
     return toIcebergTable(sparkTable);
   }
 
+  /**
+   * Returns the underlying Iceberg Catalog object represented by a Spark Catalog
+   * @param spark SparkSession used for looking up catalog reference
+   * @param catalogName The name of the Spark Catalog being referenced
+   * @return the Iceberg catalog class being wrapped by the Spark Catalog
+   */
+  public static Catalog loadIcebergCatalog(SparkSession spark, String catalogName) {
```

Comment:
@kbendick I think you would be interested in this as well

Reply:
Thanks for tagging me @RussellSpitzer. I do think this would be really beneficial. I have a number of awkward workarounds that I've either suggsted to people or implemented myself that this would get around. So emphatic +1.

Reply:
We'd probably want to add the `HasIcebergCatalog` interface to other catalogs as well, though that can be handled in a follow up once the interface is in.

If / when we merge this, let's open an issue for adding `HasIcebergCatalog` to other catalogs. I've got a newer way of tracking opened issues I've been using so that's one way to ensure that it is at least not forgotten about =)

Reply:
One issue with that is this is a Spark Table Catalog implementing class in the current iteration, should we change that?